### PR TITLE
build: add CI for distcheck

### PR DESCRIPTION
**Pull request by vtjnash**
Refs: https://github.com/libuv/libuv/issues/2190#issuecomment-465974240
Refs: https://github.com/libuv/libuv/issues/2199

### CODE REVIEW

**Comment by vtjnash** [comment]
I don't recall `tcp_close_accept` being an issue, but it does not seem related here, so should be ready to go. Note that the `distcheck` failure is ignored currently, due to a couple failing tests.

**Comment by richardlau** [comment]
> I don't recall `tcp_close_accept` being an issue, but it does not seem related here, so should be ready to go. Note that the `distcheck` failure is ignored currently, due to a couple failing tests.

FWIW a couple of years ago I attempted to run distcheck on the Jenkins CI but also ran into test failures, https://github.com/nodejs/build/issues/1910#issuecomment-584840031.

**Comment by vtjnash** [comment]
The tests depend on running from the root source directory, as we use it for reading and writing some temp files.

File: .github/workflows/CI-unix.yml [hunk 1]
```diff
@@ -14,6 +14,23 @@ on:
       - master
 
 jobs:
+  build-linux:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
```
Comment (richardlau) [inline .github/workflows/CI-unix.yml:20]:
```suggestion
      - uses: actions/checkout@v3
```

**Review by richardlau — APPROVED** [review]

**Pull request merged**